ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogAppendingCommitStrategy.scala

```diff
@@ -9,11 +9,12 @@ import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
   DamlStateKey,
   DamlStateValue
 }
-import com.daml.ledger.participant.state.kvutils.Envelope
 import com.daml.ledger.participant.state.kvutils.export.SubmissionAggregator
+import com.daml.ledger.participant.state.kvutils.{Envelope, `Bytes Ordering`}
 import com.daml.ledger.participant.state.v1.ParticipantId
+import com.daml.ledger.validator.LedgerStateOperations.{Key, Value}
 
-import scala.collection.breakOut
+import scala.collection.{SortedMap, breakOut}
 import scala.concurrent.{ExecutionContext, Future}
 
 class LogAppendingCommitStrategy[Index](
```

Comment:
You might want to add a note to `CommitStrategy` that we expect deterministic write-sets from its implementations.

Reply:
Half-cooked but maybe meaningful: how about considering (not for this PR) changing a raw write set type from `Seq[(Key, Value)]` to `SortedSet`?

Reply:
Hmmm… it doesn't really need to be a `Set`. I'd be happy with a `SortedMap` but it's quite a change.